Feat: Add Grpc support in backend

kelpikz opened the pull request.

kelpikz committed:
feat: add submodule

kelpikz committed:
feat: add grpc module

kelpikz committed:
fix: update lint and format scripts to exclude proto_build

kelpikz committed:
feat: move grpc related stuff to a seperate module grpc_manager

kelpikz committed:
feat: update Readme and gitignore

kelpikz committed:
misc: add newline at end of file(setup.cfg)

abesheknarayan commented on server/grpc-server.crt:
```diff
@@ -0,0 +1,22 @@
+xY01aSxRMQud7e8h/FDMPdTvcFpcnWUB2byRdw7gv5kIjfRsZZSwy/8zODZx/biR
+n8lG/ZKeivbUai547FHqdI2qJwv2mElxojG2hLV7sImSg26fuaTyy5+ftP3GH8Yj
+5+8T
+-----END CERTIFICATE-----
```
Add certificates under a new certs/ directory? so that we can maintain all env certificates there

kelpikz replied:
Okay

kelpikz committed:
fix:  fix typo in readme.md

kelpikz committed:
feat: add stream stub, fix cert import

abesheknarayan commented on server/app/main.py:
```diff
@@ -1,8 +1,20 @@
 
+if __name__ == "__main__":
+    grpc_manager = GrpcManager()
+    uvicorn.run("main:app", host="0.0.0.0")
```
uvicorn's default port is 8000. Run bots in another port by passing another arg port = 9000 for ex as it clashes with server's port

abesheknarayan reviewed (commented):
Tried to give a dummy login request to server, connection between bot and server seems to be ok!

kelpikz committed:
feat: add config and remove hard code for server uri and port and move it to .env

kelpikz committed:
feat: update README and add docs to config.py

kelpikz committed:
feat: move certs into a diff directory
update: update flake8 rules
misc: update core/config.py to pass the lint script

abesheknarayan commented on server/app/grpc_manager/base.py:
```diff
@@ -0,0 +1,35 @@
+            cert = open(path).read().encode("utf8")
+            creds = grpc.ssl_channel_credentials(cert)
+            channel = grpc.secure_channel(
+                "localhost:8000",
```
use config variable

kelpikz committed:
fix: fix readme formatting

abesheknarayan commented on server/app/grpc_manager/base.py:
```diff
@@ -0,0 +1,35 @@
+            # TODO: create a ping method in server to test if the connection is working
+            # login_req = LoginRequest(email="bot", password="bot")
+            # login_res = self.action_stub.Login(login_req)
+            # print("login response", login_res)
```
remove these comments before merging

abesheknarayan commented on server/.example.env:
```diff
@@ -0,0 +1,2 @@
+PORT=8001
+GRPC_SERVER_URI=https://localhost:8000
```
add newline at EOF

abesheknarayan commented on .gitignore:
```diff
@@ -1,2 +1,4 @@
\ No newline at end of file
+.venv
+.env
+proto_build
```
add newline at EOF

kelpikz committed:
fix: remove GRPC_URI hardcoding and use config instead

kelpikz committed:
fix: remove unncessary comments and add newLine at EOF for gitignore and .env.example

kelpikz committed:
fix: fix relative import
